packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js

```diff
@@ -35,6 +40,72 @@ const Indicator = () => {
   const trackedInitialLoad = useRef(false)
   const { track } = useTrackEvent()
 
+  async function fetchPageData() {
+    const urlHostString = window.location.origin
+    const pathAdjustment =
+      window.location.pathname === `/` ? `/index` : window.location.pathname
+
+    const url = `${urlHostString}/page-data${pathAdjustment}/page-data.json`
+    try {
+      const resp = await fetch(url)
+      const data = await resp.text()
+
+      if (data === `{}`) {
+        const err = new Error(`Not Found`)
+        err.status = 404
+        throw err
+      }
+
+      return { data, errorMessage: null }
+    } catch (e) {
+      if (e.status === 404) {
+        return { data: null, errorMessage: `This page has moved.` }
+      } else {
+        return { data: null, errorMessage: null }
+      }
+    }
+  }
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const { data } = await fetchPageData()
+      pageData = data
+    }
+
+    fetchData()
+  }, [])
+
+  const hasPageDataChanged = async () => {
+    if (buildId !== latestCheckedBuild || !pageData) {
+      let pageDataCounter = 0
+      const hasPageChanged = false
```

Comment:
should this be a `let`? Did you mean to reassign this variable on line 93 rather than initialize a brand new variable?

Reply:
good catch!